src/bucket.js

```diff
@@ -97,6 +98,7 @@ Bucket.prototype.createSignedGetUrl = function(prefix, options) {
   assert(prefix,                'prefix must be given');
   assert(options,               'options must be given');
   assert(options.expires,       'expires must be given');
+  prefix = encodeURIComponent(prefix);
```

Comment:
Can we please avoid escaping anything that doesn't strictly need to be escaped...

Notably, this will cause `/` to be escaped.. let's try to avoid that.

Reply:
I'm very reluctant to not fully URI encode things, since we're specifically trying to fix something where we made incorrect assumptions about how keys are URI encoded within S3.

Since these are not queue urls, there's no real reason to make them look pretty, only risk.  The queue urls would still have `/` in them, just that the redirect `Location` headers won't.  Anyone looking at the redirect `Location` likely understands enough to be able to figure out that they're URI encoded.

Reply:
@jhford, fair point, but please monitor the trees and make an announcement about the risk with this encoding. I'm not convinced that encoding `/` won't cause problems.

Reply:
@djmitche, can you help make a call on the risk of encoding `/` in the URLs that the queue redirects to...

it's plausible that I'm being irrational here :)

Reply:
I think the risk is low, but the reward is so small that it's still not worth it.  Let's deploy this an revisit if we have a lot of complaints about the urls in the `Location` headers

Reply:
I don't know what the original encoding issue was with S3 -- @jhford says "S3 will URL Encode some characters of the prefix" in the intro to this PR, but I don't know when S3 generates URLs..

It's not surprising that S3 accepts "partial URI encoding" -- because URI encoding is identity for so many characters, it can be a funny encoding to deal with.  S3 just decodes any %xx's that it finds, so I'm sure it would happily decode `%4A` to `J`.  That's not really unusual.  Regarding `/` specifically, the S3 backend doesn't treat the character specially, and it doesn't expect any other path components in the key name, so it's just as happy with `%2F` as with `/`.  To "fully" urlencode something would be to produce a string of only `%xx%xx%xx`, and I don't think we should do that.  The question is what characters to encode -- and that needs to be at least the set of characters which the receiving system (S3, in this case?) would interpret as having another meaning.  Anything over that set is unnecessary but probably won't cause issues with that receiving system.

In other words, I don't think it matters to S3.  And it doesn't matter much to users, since they generally won't see these URLs.  History suggests it probably will matter to @emorley though.  And for my own purposes, it's nice to be able to read the URLs in a logfile, and `/foo` is a lot easier to read than `%2Ffoo`.